neuralop/utils.py

```diff
@@ -64,78 +69,128 @@ def to(self, device):
     """This function computes the spectrum of a 2D signal using the Fast Fourier Transform (FFT).
 
     Paramaters
     ----------
-    signal : a tensor of shape (T * n_observations * n_observations) 
-        A 2D discretized signal represented as a 1D tensor with shape (T * n_observations * n_observations), where T is the number of time steps and n_observations is the spatial size of the signal. 
-        T can be any number of channels that we reshape into and n_observations * n_observations is the spatial resolution.
+    signal : a tensor of shape (T * n_observations * n_observations)
+        A 2D discretized signal represented as a 1D tensor with shape
+        (T * n_observations * n_observations), where T is the number of time
+        steps and n_observations is the spatial size of the signal.
+
+        T can be any number of channels that we reshape into and
+        n_observations * n_observations is the spatial resolution.
     n_observations: an integer
         Number of discretized points. Basically the resolution of the signal.
-        
+
     Returns
     --------
-    spectrum: a tensor 
+    spectrum: a tensor
         A 1D tensor of shape (s,) representing the computed spectrum.
     """
     T = signal.shape[0]
     signal = signal.view(T, n_observations, n_observations)
-    
+
     if normalize:
         signal = torch.fft.fft2(signal)
     else:
-        signal = torch.fft.rfft2(signal, s=(n_observations, n_observations), normalized=False)
-    
+        signal = torch.fft.rfft2(
+            signal, s=(n_observations, n_observations), normalized=False
+        )
+
     # 2d wavenumbers following PyTorch fft convention
     k_max = n_observations // 2
-    wavenumers = torch.cat((torch.arange(start=0, end=k_max, step=1), \
-                            torch.arange(start=-k_max, end=0, step=1)), 0).repeat(n_observations, 1)
+    wavenumers = torch.cat(
+        (
+            torch.arange(start=0, end=k_max, step=1),
+            torch.arange(start=-k_max, end=0, step=1),
+        ),
+        0,
+    ).repeat(n_observations, 1)
     k_x = wavenumers.transpose(0, 1)
     k_y = wavenumers
-    
+
     # Sum wavenumbers
     sum_k = torch.abs(k_x) + torch.abs(k_y)
     sum_k = sum_k
-    
+
     # Remove symmetric components from wavenumbers
     index = -1.0 * torch.ones((n_observations, n_observations))
-    index[0:k_max + 1, 0:k_max + 1] = sum_k[0:k_max + 1, 0:k_max + 1]
-    
+    index[0 : k_max + 1, 0 : k_max + 1] = sum_k[0 : k_max + 1, 0 : k_max + 1]
+
     spectrum = torch.zeros((T, n_observations))
     for j in range(1, n_observations + 1):
         ind = torch.where(index == j)
         spectrum[:, j - 1] = (signal[:, ind[0], ind[1]].sum(dim=1)).abs() ** 2
-    
+
     spectrum = spectrum.mean(dim=0)
     return spectrum
 
+
+Number = Union[float, int]
+
+def validate_output_scaling_factor_1d(
+    output_scaling_factor: Optional[Union[Number, List[Number]]],
+    order: int,
```

Comment:
Why do you call them 1d and 2d? They are both N-D, where order = n_dim = N. 
Given this I think we should merge them and just have n_layers=1 by default, what do you think?

Reply:
> Why do you call them 1d and 2d?
That's just based on their return value. `validate_*_2d` returns a list of shapes, while `validate_*_1d` returns just a shape.

> I think we should merge them and just have n_layers=1 by default
That sounds fine, as long as a singleton "list" of shapes doesn't confuse any of the models.

Reply:
I see - how about we call them validate_output_scaling_factor_single_layer and validate_output_scaling_factor_multiple_layers? 

A mouthful and a pain to write but self-explanatory and never used by end users most likely?

Reply:
Ideally we should never have to do both, makes me think there is a deficiency in the code - we should either check at model level or layer level, probably not both

Reply:
I made an allowance for non-positive values of `n_layers` to return a single layer instead of a list of layers (for a whole module). This is only used in `DomainPadding`.